### PR TITLE
Add "platforms" to a few API calls

### PR DESCRIPTION
**Pull request by raskchanky**
tenor-214332903

Closes https://github.com/habitat-sh/habitat/issues/2973

Signed-off-by: Josh Black <raskchanky@gmail.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-originsrv/src/data_store.rs [hunk 1]
```diff
@@ -692,12 +692,17 @@ impl DataStore {
 
             let release_count: i64 = row.get("release_count");
             let latest: String = row.get("latest");
+            let platforms_str: String = row.get("platforms");
+            let platforms_vec = platforms_str.split(',').map(|x| x.to_string()).collect();
+            let platforms = protobuf::RepeatedField::from_vec(platforms_vec);
```
Comment (chefsalim) [inline components/builder-originsrv/src/data_store.rs]: ah, nice use of the from_vec

File: components/builder-depot/src/server.rs [hunk 2]
```diff
@@ -2549,7 +2569,7 @@ mod test {
             },
             "checksum":"checksum",
             "manifest":"manifest",
-            "target":"x86_64-linux",
+            "platform":"x86_64-linux",
```
Comment (chefsalim) [inline components/builder-depot/src/server.rs]: We seem to be using the terminology for Target to mean Platform + Architecture... do we want to be changing that here?

**Comment by chefsalim** [comment]
LGTM other than the comment on target vs. platform

File: components/builder-depot/src/server.rs (hunk 2 above)
Comment (raskchanky) [reply components/builder-depot/src/server.rs]: I was just trying to be consistent, but I see your point.  I'll change it back.

**Comment by raskchanky** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/269703371).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/269703371).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**